### PR TITLE
service discovery: Add check configuration by default

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

Adds service discovery check by default

Moves configuration check to Run, instead of Configure

reverts #29033 & fixes the issue on the CI

### Motivation

Ease running the check

### CODE REVIEW

**Review by Kaderinho — APPROVED** [review]

**Review by Pythyu — APPROVED** [review]

File: pkg/collector/corechecks/servicediscovery/servicediscovery.go [hunk 1]
```diff
@@ -137,6 +134,10 @@ func (c *Check) Configure(senderManager sender.SenderManager, _ uint64, instance
 
 // Run executes the check.
 func (c *Check) Run() error {
+	if !pkgconfig.SystemProbe().GetBool("discovery.enabled") {
+		return errors.New("service discovery is disabled")
```
[the remark this answers is not in the document]
Comment (guyarb) [reply pkg/collector/corechecks/servicediscovery/servicediscovery.go:138]: sorry, I should have return nil

File: test/new-e2e/tests/agent-subcommands/configcheck/configcheck_nix_test.go [hunk 2]
```diff
@@ -82,6 +82,12 @@ func (v *linuxConfigCheckSuite) TestDefaultInstalledChecks() {
 			InstanceID: "uptime:",
 			Settings:   "{}",
 		},
+		{
+			CheckName:  "service_discovery",
+			Filepath:   "file:/etc/datadog-agent/conf.d/service_discovery.d/conf.yaml.default",
```
[the remark this answers is not in the document]
Comment (guyarb) [reply test/new-e2e/tests/agent-subcommands/configcheck/configcheck_nix_test.go:87]: WDYM? 

I added this as I do want to have the configuration created by default

[the remark this answers is not in the document]
Comment (vitkyrka) [reply test/new-e2e/tests/agent-subcommands/configcheck/configcheck_nix_test.go:87]: I meant that this test can also stop installing `config.yaml`: https://github.com/DataDog/datadog-agent/blob/main/test/new-e2e/tests/discovery/linux_test.go#L45

**Review by vitkyrka — APPROVED** [review]

**Pull request merged**